File: back/core/src/main/java/com/bird/maru/notice/repository/NoticeRepository.java
package com.bird.maru.notice.repository;

import com.bird.maru.notice.model.Notice;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface NoticeRepository extends MongoRepository<Notice, String> {

    Slice<Notice> findByMemberIdOrderByCreatedDateTimeDesc(Long memberId, Pageable pageable);

}

File: front/app/src/main/java/com/shoebill/maru/model/repository/NoticeRepository.kt
package com.shoebill.maru.model.repository

import com.shoebill.maru.model.data.notice.Notice
import com.shoebill.maru.model.interfaces.NoticeApi
import retrofit2.Response
import javax.inject.Inject

class NoticeRepository @Inject constructor(
    private val noticeApi: NoticeApi,
) {
    suspend fun getNotices(page: Int = 0): Response<List<Notice>> = noticeApi.getNotices(page)
}